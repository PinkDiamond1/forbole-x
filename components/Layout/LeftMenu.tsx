import React from 'react'
import useTranslation from 'next-translate/useTranslation'
import OverviewIcon from '../../assets/images/icons/icon_overview.svg'
import WalletManageIcon from '../../assets/images/icons/icon_wallet_manage.svg'
import DelegateIcon from '../../assets/images/icons/icon_delegate_08.svg'
import MarketIcon from '../../assets/images/icons/icon_market.svg'
import ProposalIcon from '../../assets/images/icons/icon_proposal.svg'
import AddressBookIcon from '../../assets/images/icons/icon_address_book.svg'
import ExplorerIcon from '../../assets/images/icons/icon_explorer.svg'
import Logo from '../../assets/images/logo.svg'
import { useTheme } from '@material-ui/core/styles'
import { List, ListItem, ListItemIcon, ListItemText, Paper } from '@material-ui/core'
import useStyles from './styles'
import Link from 'next/link'

interface LeftMenuProps {
  activeItem: string
}

const LeftMenu: React.FC<LeftMenuProps> = ({ activeItem }) => {
  const theme = useTheme()
  const { t } = useTranslation('common')
  const iconProps = {
    width: theme.spacing(3),
    height: theme.spacing(3),
  }
  const classes = useStyles()
  const items = React.useMemo(
    () => [
      {
        title: t('overview'),
        icon: <OverviewIcon {...iconProps} />,
        href: '/',
      },
      {
        title: t('walletManage'),
        icon: <WalletManageIcon {...iconProps} />,
        href: '/wallets',
      },
      {
        title: t('delegate'),
        icon: <DelegateIcon {...iconProps} />,
        href: '/delegate',
      },
      {
        title: t('market'),
        icon: <MarketIcon {...iconProps} />,
        href: '/market',
      },
      {
        title: t('proposals'),
        icon: <ProposalIcon {...iconProps} />,
        href: '/proposals',
      },
      {
        title: t('addressBook'),
        icon: <AddressBookIcon {...iconProps} />,
        href: '/address-book',
      },
      {
        title: t('explorer'),
        icon: <ExplorerIcon {...iconProps} />,
        href: '/explorer',
      },
    ],
    [iconProps, t]
  )

  return (
    <Paper elevation={0} className={classes.leftMenuContainer}>
      <List className={classes.menu}>
        <ListItem className={classes.menuItem} button>
          <ListItemIcon>
            <Logo
              width={theme.spacing(5)}
              height={theme.spacing(5)}
              fill={theme.palette.primary.main}
            />
          </ListItemIcon>
          <ListItemText
            primary="PORTAL"
            primaryTypographyProps={{
              variant: 'h1',
              color: 'primary',
            }}
          />
        </ListItem>
        {items.map((item) => {
          const selected = item.href === activeItem
          return (
            <Link key={item.title} href={item.href} passHref>
              <ListItem selected={selected} className={classes.menuItem} button component="a">
                <ListItemIcon>
                  {React.cloneElement(item.icon, {
                    fill: selected ? theme.palette.primary.main : theme.palette.grey[300],
                  })}
                </ListItemIcon>
                <ListItemText
                  primary={item.title}
                  primaryTypographyProps={{
                    variant: 'h6',
                    color: selected ? 'primary' : 'textSecondary',
                  }}
                />
              </ListItem>
            </Link>
          )
        })}
      </List>
    </Paper>
  )
}

export default React.memo(LeftMenu)
