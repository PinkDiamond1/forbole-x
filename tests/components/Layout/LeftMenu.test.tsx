import React from 'react'
import renderer from 'react-test-renderer'
import LeftMenu from '../../../components/Layout/LeftMenu'

describe('component: Layout - LeftMenu', () => {
  it('renders no matching activeItem correctly', () => {
    const component = renderer.create(<LeftMenu activeItem="" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders activeItem "/" correctly', () => {
    const component = renderer.create(<LeftMenu activeItem="/" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
