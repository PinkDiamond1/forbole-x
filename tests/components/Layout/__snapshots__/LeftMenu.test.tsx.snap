// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Layout - LeftMenu renders activeItem "/" correctly 1`] = `
<div
  className="MuiPaper-root HookGlobalStyles-leftMenuContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "width": 256,
    }
  }
>
  <ul
    className="MuiList-root HookGlobalStyles-menu-3 MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <div
          className="MuiBox-root MuiBox-root-16"
        >
          <svg
            data-file-name="SvgLogo_expended"
            fill="#3f51b5"
          />
        </div>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-17"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button Mui-selected"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": undefined,
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_overview"
            fill="#3f51b5"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-displayBlock"
          >
            common:portfolio
          </span>
        </div>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/wallets"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_wallet_manage"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:wallet
          </span>
        </div>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/delegate"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_delegate_08"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:delegate
          </span>
        </div>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/address-book"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_address_book"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:address book
          </span>
        </div>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/proposals"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_proposal"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:proposals
          </span>
        </div>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
    </div>
    <div
      className="MuiBox-root MuiBox-root-18 HookGlobalStyles-favMenu-5"
    >
      <div
        className="MuiListItemText-root HookGlobalStyles-starredAccounts-6"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          common:starredAccounts
        </span>
      </div>
      <div
        className="MuiListItemText-root HookGlobalStyles-manageAccounts-7"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          common:manageAccounts
        </span>
      </div>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-starButton-8 MuiButton-containedPrimary"
        href="/wallets"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          common:star now
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
    </div>
  </ul>
</div>
`;

exports[`component: Layout - LeftMenu renders no matching activeItem correctly 1`] = `
<div
  className="MuiPaper-root HookGlobalStyles-leftMenuContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "width": 256,
    }
  }
>
  <ul
    className="MuiList-root HookGlobalStyles-menu-3 MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <div
          className="MuiBox-root MuiBox-root-13"
        >
          <svg
            data-file-name="SvgLogo_expended"
            fill="#3f51b5"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-14"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_overview"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:portfolio
          </span>
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/wallets"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_wallet_manage"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:wallet
          </span>
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/delegate"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_delegate_08"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:delegate
          </span>
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/address-book"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_address_book"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:address book
          </span>
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root HookGlobalStyles-menuItem-4 MuiListItem-gutters MuiListItem-button"
        href="/proposals"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "inherits",
          }
        }
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            data-file-name="SvgIcon_proposal"
            fill="#e0e0e0"
            height={24}
            width={24}
          />
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            common:proposals
          </span>
        </div>
      </a>
    </div>
    <div
      className="MuiBox-root MuiBox-root-15 HookGlobalStyles-favMenu-5"
    >
      <div
        className="MuiListItemText-root HookGlobalStyles-starredAccounts-6"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          common:starredAccounts
        </span>
      </div>
      <div
        className="MuiListItemText-root HookGlobalStyles-manageAccounts-7"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-h6 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          common:manageAccounts
        </span>
      </div>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-starButton-8 MuiButton-containedPrimary"
        href="/wallets"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          common:star now
        </span>
      </a>
    </div>
  </ul>
</div>
`;
